lib/gen_stage.ex

```diff
@@ -1746,6 +1765,7 @@ defmodule GenStage do
         type: :producer,
         buffer: Buffer.new(buffer_size),
         buffer_keep: buffer_keep,
+        mod_handle_discarded: function_exported?(mod, :handle_discarded, 2),
```

Comment:
We don't need to cache it. `function_exported?` checks are performance, we can simply check it before calling.

Reply:
Right, this was only intended as a (premature) performance optimization. I'll remove it.